include/boost/graph/detail/adjacency_list.hpp

```diff
@@ -2069,15 +2069,20 @@ namespace detail
     inline void reindex_edge_list(
         EdgeList& el, vertex_descriptor u, boost::disallow_parallel_edge_tag)
     {
-        for (typename EdgeList::iterator ei = el.begin(); ei != el.end(); ++ei)
+        typename EdgeList::iterator ei = el.begin(), e_end = el.end();
+        while (ei != e_end)
         {
             if (ei->get_target() > u)
             {
                 typename EdgeList::value_type ce = *ei;
+                ++ei;
                 el.erase(ce);
```

Comment:
```suggestion
                ei = el.erase(ce);
```

Reply:
Thanks for the suggestion. It indeed looks like a better approach; however, I was not able to get this code to compile:

```
#include <boost/graph/adjacency_list.hpp>

using Graph = boost::adjacency_list<boost::setS, boost::vecS, boost::undirectedS>;

int main() {
  Graph g(3);
  boost::add_edge(2, 0, g);
  boost::remove_vertex(1, g);
}
```

The gist of the error is that `erase` does not return an iterator in this case, but a `long unsigned int`.

(The original patch was simply a reversion of a previous commit.)